src/widgets/Forms/AddExpenseForm/AddExpenseCalc.tsx

```diff
@@ -143,17 +143,17 @@ const AddExpenseCalculator = ({ closeModal }: TExpensesCalculatorProps) => {
     setAmount((prevValue) => {
       if (value === 'del') {
         return prevValue.length === 1 ? '0' : prevValue.slice(0, -1);
+      } else if (value === 'Clear') {
+        setSelectedPaymentSource('');
+        setSelectedCategory('');
+
+        return '0';
       } else if (value !== '.' && prevValue === '0') {
         return value;
       } else if (value === '.' && prevValue.includes('.')) {
         return prevValue;
       } else if (prevValue.split('.')[1]?.length >= 2) {
         return prevValue;
-      } else if (value === 'Clear') {
```

Comment:
position is important?

Reply:
yes, cause if we get in first if we are not going in other...